src/epics/handleNightModeToggle.js

```diff
@@ -1,26 +1,100 @@
+import gql from 'graphql-tag'
 import { Observable } from 'rxjs'
 import {
   USER_TOGGLE_NIGHT_MODE,
-  APP_USER_NIGHT_MODE_SAVE
+  APP_USER_NIGHT_MODE_SAVE,
+  APP_USER_NIGHT_MODE_SAVE_ERROR,
+  CHANGE_USER_DATA
 } from './../actions/types'
-import { saveKeyState } from '../utils/localStorage'
+import { saveKeyState, loadKeyState } from '../utils/localStorage'
+import { handleErrorAndTriggerAction } from './utils'
 
-const handleNightModeToggle = action$ =>
+const NIGHT_MODE_MUTATION = gql`
+  mutation updateUserSettings($theme: String!) {
+    updateUserSettings(settings: { theme: $theme }) {
+      theme
+    }
+  }
+`
+
+export const THEME_TYPES = {
+  default: 'default',
+  nightmode: 'nightmode'
+}
+
+const handleNightModeToggle = (action$, store, { client }) =>
   action$
     .ofType(USER_TOGGLE_NIGHT_MODE)
     .exhaustMap(() =>
       Observable.of(document.body.classList.toggle('night-mode'))
     )
-    .debounceTime(200)
-    .map(isNightModeEnabled => {
-      saveKeyState('isNightModeEnabled', isNightModeEnabled)
-      return Observable.of(isNightModeEnabled)
+    .mergeMap(isNightModeEnabled => {
+      saveKeyState('isNightMode', isNightModeEnabled)
+      const { data } = store.getState().user
+      if (data && data.id) {
+        const mutation = client.mutate({
+          mutation: NIGHT_MODE_MUTATION,
+          variables: {
+            theme: isNightModeEnabled
+              ? THEME_TYPES.nightmode
+              : THEME_TYPES.default
+          }
+        })
+        return Observable.from(mutation)
+          .mergeMap(({ data: { updateUserSettings } }) => {
+            return Observable.of({
+              type: APP_USER_NIGHT_MODE_SAVE,
+              payload: updateUserSettings.theme === THEME_TYPES.nightmode
+            })
+          })
+          .catch(handleErrorAndTriggerAction(APP_USER_NIGHT_MODE_SAVE_ERROR))
+      } else {
+        return Observable.of({
+          type: APP_USER_NIGHT_MODE_SAVE,
+          payload: isNightModeEnabled
+        })
+      }
     })
-    .mergeMap(({ value }) =>
-      Observable.of({
+
+export const saveNightModeAfterLaunch = action$ =>
+  action$
+    .ofType(CHANGE_USER_DATA)
+    .filter(
+      ({ user = {} }) =>
+        user.settings && user.settings.theme === THEME_TYPES.nightmode
+    )
+    .filter(() => loadKeyState('isNightMode') === undefined)
+    .exhaustMap(() => Observable.of(document.body.classList.add('night-mode')))
+    .mergeMap(() => {
+      saveKeyState('isNightMode', true)
+      return Observable.of({
         type: APP_USER_NIGHT_MODE_SAVE,
-        payload: value
+        payload: true
       })
+    })
+
+export const sendNightModeIfDiff = (action$, store, { client }) =>
+  action$
+    .ofType(CHANGE_USER_DATA)
+    .filter(
+      () =>
+        loadKeyState('isNightMode') === undefined ||
+        loadKeyState('isNightModeEnabled') === true
```

Comment:
If it's possible, could we take the value of the `isNightMode` from our "single source of truth" - redux store?
Redux store is initialised with a [`LS` value](https://github.com/santiment/app/pull/457/files?file-filters%5B%5D=.js#diff-e6c10270750f429a2a3409d7f330a028R4), so the setting value should be correct after the app has launched.

Reply:
Here the case, when previous `LS` field (`isNightModeEnabled`) contain true statement and new field haven't determined yet (`isNightMode`) and we need to upd settings on server for user (for preventing diff between predefined `night mode` in LS and `default` on server after merge this PR). It's for consistency. 
In store we have only one field for night mode, that can contain value from any these two fields.
I can create two additional fields in store, but...I think it will be more confusion...I can write a NOTE here for what I use this epic.
___
After your review I also found a mistake, that here should be `&&` instead `||`. Thank you)